plugin/js/parsers/JadeScrollsParser.js

```diff
@@ -0,0 +1,443 @@
+                if (links.length > 0) {
+                    chapters = util.hyperlinksToChapterList(links);
+                } else if (baseUrl.includes("/episode-")) {
+                    // Single chapter fallback
+                    return [{
+                        sourceUrl: baseUrl,
+                        title: "Chapter"
+                    }];
+                }
+            }
+
+            // Sort by episode number to ensure correct order
+            chapters.sort((a, b) => {
+                let numA = this.extractEpisodeNumber(a.sourceUrl);
+                let numB = this.extractEpisodeNumber(b.sourceUrl);
+                return numA - numB;
+            });
+
+            return chapters;
+        } catch (error) {
+            // If API fails, fall back to DOM parsing
+            console.error("JadeScrolls API fetch failed, falling back to DOM parsing:", error);
+            let links = [...dom.querySelectorAll("a[href*='/episode-']")];
+            return util.hyperlinksToChapterList(links);
+        }
+    }
+
+    // Extract novel slug from URL
+    extractNovelSlug(url) {
+        let match = url.match(/\/novel\/([^/]+)/);
+        return match ? match[1] : null;
+    }
+
+    // Extract the novel base URL from a chapter URL
+    getNovelUrl(url) {
+        // Convert /novel/{slug}/episode-{n} to /novel/{slug}
+        let match = url.match(/(https?:\/\/[^/]+\/novel\/[^/]+)/);
+        return match ? match[1] : url;
+    }
+
+    // Extract episode/chapter number from URL for sorting
+    extractEpisodeNumber(url) {
+        // Try episode-N format
+        let match = url.match(/episode-(\d+)/);
+        if (match) {
+            return parseInt(match[1], 10);
+        }
+
+        // Try chapter-N format
+        match = url.match(/chapter-(\d+)/);
+        if (match) {
+            return parseInt(match[1], 10);
+        }
+
+        // Fallback: extract any number from the slug
+        match = url.match(/(\d+)/);
+        return match ? parseInt(match[1], 10) : 0;
+    }
+
+    // Extract the chapter title from the page
+    extractChapterTitle(dom) {
+        let titleElement = dom.querySelector(".chpt-title.fw-medium");
+        return titleElement ? titleElement.textContent.trim() : null;
+    }
+
+    // Find the content element on the chapter page
+    findContent(dom) {
+        // Look for the main content container
+        let content = dom.querySelector(".content-inner.listing-disc-styling");
+
+        if (!content) {
+            // Content not found - this happens when the site uses JS to load content
+            // Try to find any content container that might exist
+            content = dom.querySelector("div.content-inner") ||
+                     dom.querySelector("div[class*='content']");
+        }
+
+        return content;
+    }
+
+    // Extract chapter title (called separately from content)
+    findChapterTitle(dom) {
```

Comment:
findChapterTitle() is not needed, because we're constructing a "fetched" chapter, we've already put the title in the content.